nullaway/src/test/java/com/uber/nullaway/NullAwaySerializationTest.java

```diff
@@ -41,21 +46,48 @@
 /** Unit tests for {@link com.uber.nullaway.NullAway}. */
 @RunWith(JUnit4.class)
 public class NullAwaySerializationTest extends NullAwayTestsBase {
-
-  private SerializationTestHelper serializationTestHelper;
   private String configPath;
+  private Path root;
```

Comment:
@msridhar The output files these test read are controlled by the path passed when constructing the `SerializationTestHelper<>` object plus the file passed to `.expectedOutputs(...)`. For most of these tests, that is `${this.root}/${this.SUGGEST_FIX_FILE_NAME}`. 

That means these tests can't ever run in parallel. Just out of curiosity, is there any guarantee for gradle/JUnit such as: different test classes run in parallel, but test cases within the same `@RunWith` annotated class run serially?

That would make sense to me, and would mean this is correct. But I can't find no such guarantee in the gradle parallel testing docs, just a warning that:

> When using parallel test execution, make sure your tests are properly isolated from one another. Tests that interact with the filesystem are particularly prone to conflict, causing intermittent test failures.

- https://docs.gradle.org/current/userguide/java_testing.html

Reply:
@lazaroclapp AFAIK Gradle will not run test methods within the same test class in parallel with each other.  It only runs test methods from different test classes in parallel.  This could cause a problem down the road if we split this test into multiple tests I guess.  Could we fix by using a unique temp (sub-)directory `root` in every test?

Reply:
This would work even if we split the test into multiple classes, because each such class will likely have its own `private Path root` instance. My only worry here is if there is some aggressively parallel JUnit+Gradle mode that runs a unit tests within the same test class in parallel, if you don't think that's likely (and it would break a lot of `setUp(...)` methods I've seen), then this is probably fine as it is.

Reply:
I am assuming the latest decision is to leave this as it is. In that case this is ready for another round of review.

Reply:
Looking more, I think the way `@Rule` works, the temporary folder should already get created fresh for each test and deleted afterward.  Due to how these things work, I don't think any test runner would ever attempt to run tests within a class in parallel.  So I think this code is fine as-is.  That said, I did find another change we should make that I will comment on separately.